### DISCUSSION TITLE
status of this project?

### DISCUSSION THREAD

**Discussion by hamburger-fries** [polonel/trudesk#581] [General]
We are about to build something for our company using NextJS. But found this interesting project.
Perhaps we can expand the API and related code to work with NextJS + NextAUTH.

Also, we would love to help rewrite the entire app with Typescript.

**Comment by shahyar** [comment]
Hey there. We're already building v2 of Trudesk. The API is way more extensible, and uses GraphQL. We'll be releasing our cloud-based version first very early 2023, alongside some SDK packages for JS/TS and React. The open-source API server will come later; we're aiming mid-year for that.

New client and SDKs are TypeScript, as well. Server is currently being migrated to TS.

**Reply by cameronkilton** [reply]
Interested in helping accelerate the open source API server? Is the new front end not going to be available?

**Reply by shahyar** [reply]
> Interested in helping accelerate the open source API server? 

We'd love to speed things up, but we're a small team. Our main priorities are to launch Trudesk Cloud, and ship the Trudesk On-Premises images, as a couple of partners depend on it. I do say mid-2023, but we're aiming for faster.

> Is the new front end not going to be available?

It will, it's just a matter of when. As it's completely written from scratch, and therefore depends on the GraphQL API, it can't be used on a v1 server. If it comes out before server v2, it would only be usable against the cloud API endpoints.

It's important for us that we really nail the server implementation, because we want people to be able to use the SDK to integrate Trudesk in any way they need on their existing software and platforms.

**Reply by cameronkilton** [reply]
It was a statement. I'm interested in helping augment your team with additional folks at my expense. I have folks with the expieriance you would need to move it forward.
